packages/notebook/src/browser/service/notebook-model-resolver-service.ts

```diff
@@ -0,0 +1,66 @@
+// *****************************************************************************
+// Copyright (C) 2023 TypeFox and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { URI } from '@theia/core';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { Emitter } from '@theia/core/shared/vscode-languageserver-protocol';
+import { UriComponents } from '@theia/core/lib/common/uri';
+import { FileService } from '@theia/filesystem/lib/browser/file-service';
+import { NotebookData } from '../../common';
+import { NotebookModel } from '../view-model/notebook-model';
+import { NotebookService } from './notebook-service';
+
+@injectable()
+export class NotebookModelResolverService {
+
+    @inject(FileService)
+    protected fileService: FileService;
+
+    @inject(NotebookService)
+    protected notebookService: NotebookService;
+
+    private onDidChangeDirtyEmitter = new Emitter<NotebookModel>();
```

Comment:
because this emitter is a general one for if any notebook is changed. Also we can't inject the models globally. They only exist within their own container